### PR TITLE
[Fix](Cooldown) fixed table creation failure caused by abnormal coold…

### LINKED ISSUE #32366: [Bug] storage policy 设置 cooldown_datetime 问题
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/doris/issues?q=is%3Aissue) and found no similar issues.

### Version

2.1.0

### What's Wrong?

```
1、创建了存储策略
create storage policy test3
PROPERTIES(
  "storage_resource" = "costest",
  "cooldown_datetime" = "2024-06-01 00:00:00"
);

> show storage policy;
+------------+-------+---------+---------+-----------------+---------------------+-------------+
| PolicyName | Id    | Version | Type    | StorageResource | CooldownDatetime    | CooldownTtl |
+------------+-------+---------+---------+-----------------+---------------------+-------------+
| test3      | 20337 | 0       | STORAGE | costest         | 2024-06-01 00:00:00 | -1          |
+------------+-------+---------+---------+-----------------+---------------------+-------------+

2、修改存储策略
alter storage policy test3
PROPERTIES(
  "cooldown_datetime" = "1000-01-01 00:00:00",
  "cooldown_ttl" = "1d"
);

3、查看存储策略
> show storage policy;
+------------+-------+---------+---------+-----------------+------------------+-------------+
| PolicyName | Id    | Version | Type    | StorageResource | CooldownDatetime | CooldownTtl |
+------------+-------+---------+---------+-----------------+------------------+-------------+
| test3      | 20337 | 1       | STORAGE | costest         | NULL             | 86400       |
+------------+-------+---------+---------+-----------------+------------------+-------------+

4、创建表报错
CREATE TABLE IF NOT EXISTS test_table (
	id INT,
	value text,
	ts date
)
	ENGINE=OLAP
	DUPLICATE KEY(id)
	DISTRIBUTED BY RANDOM
	PROPERTIES (
	"replication_num" = "1",
	"storage_policy" = "test3"
);

ERROR 1105 (HY000): errCode = 2, detailMessage = errCode = 2, detailMessage = remote storage cool down time: -30610252800000 should later than now: 1710732836426
```

从报错信息来看，迁移策略还存在cooldown_datetime，与show storage policy看的不一样。

### What You Expected?

修正存储策略判断逻辑？能够使用修改的存储策略创建表。

### How to Reproduce?

_No response_

### Anything Else?

_No response_

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by felixwluo**
## Proposed changes

Issue Number: close #32366

`When creating and modifying cold policies. It determines whether it is less than the current time, and if it is less than the current time, it throws an exception`

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by felixwluo** [comment]
run buildall

**Comment by felixwluo** [comment]
run buildall

**Comment by ByteYue** [comment]
Could you please also note this behavior change on the Documentation?

**Comment by ByteYue** [comment]
And please also add one FE UT to test this.

File: fe/fe-core/src/main/java/org/apache/doris/common/util/PropertyAnalyzer.java [hunk 1]
```diff
@@ -305,9 +305,7 @@ public static DataProperty analyzeDataProperty(Map<String, String> properties, f
             // check remote storage cool down timestamp
             if (storagePolicy.getCooldownTimestampMs() != -1) {
                 if (storagePolicy.getCooldownTimestampMs() <= currentTimeMs) {
-                    throw new AnalysisException(
```
Comment (yiguolei) [inline fe/fe-core/src/main/java/org/apache/doris/common/util/PropertyAnalyzer.java:308]: Do not modify it like this.
1. Should check the property when create or alter the storage policy if the cooldown time is less than now.
2. Throw exception when create table if the cooldown time is less than now.

**Review by yiguolei — CHANGES_REQUESTED** [review]
We should not modify the value that user set.
Should check the cooldown time is valid when create or alter storage policy.
Must throw exception when create table if the cooldown time is not valid.

**Comment by felixwluo** [comment]
> We should not modify the value that user set. Should check the cooldown time is valid when create or alter storage policy. Must throw exception when create table if the cooldown time is not valid.

Ok. I'll modify it

**Comment by felixwluo** [comment]
run buildall

**Pull request closed without merging**